### PR TITLE
Simpler and bug free loop for parsing erp response

### PR DESCRIPTION
**Pull request by yashsirvi**
- The previous loop was too convoluted and not easy to follow.
- Had a bug of not adding the last lecture of a day because the statement of adding the previous course at the end of the day was unreachable.
    - Put that code outside the loop of the cells (hours) and instead at the end of each iteration of the row loop (days).

It works with my timetable: Empty work days and variable length classes, but please verify once with other timetables.
sorry for the last two pr they had extra files I made for debugging :p

### CODE REVIEW

File: timetable/extractor.py [hunk 1]
```diff
@@ -63,56 +63,47 @@ def create_timings(_table: Tag | NavigableString) -> list[int]:
...
         day_cell: PageElement = row.find('td', {'valign': 'top'})
         if not day_cell:
             continue
-        day = days[day_cell.get_text()] if day_cell.get_text() in days.keys() else day_cell.get_text()
-
-        # Merge timeslots occurring adjacent to each other and initialize course objects
+        day = days.get(day_cell.get_text(), day_cell.get_text())
+    
         prev: Course | None = None
         course_duration: int = 0
```
Comment (harshkhandeparkar) [inline timetable/extractor.py:74]: This code was removed iirc. Is your branch up to date?

Comment (yashsirvi) [reply timetable/extractor.py:74]: Ah no it wasn't up to date. End of day bug is solved then.

Comment (harshkhandeparkar) [reply timetable/extractor.py:74]: If your improvements still simplify the code, please update the branch, and I'll merge it.

Comment (yashsirvi) [reply timetable/extractor.py]: I have updated the branch. Please check once.

File: timetable/extractor.py [hunk 2]
```diff
@@ -73,60 +73,43 @@ def build_courses(html: str, course_names: dict) -> list[Course]:
...
+    
         prev: Course | None = None
         cells = [cell for cell in row.find_all('td') if cell.attrs.get('valign') != 'top']
-
+    
         for index, cell in enumerate(cells):
-            # Empty cell means either previous class just ended or no class is scheduled
-            if cell.get_text().strip() == "":
+            text = cell.get_text().strip()
+            if not text or text in [' ', ' ']: # encountered empty cell: either commit the prev course or skip
```
Comment (harshkhandeparkar) [inline timetable/extractor.py:89]: The `.strip()` method seems to remove those two special characters as well. Can the statement be repaced with just `if not text`?

File: timetable/extractor.py [hunk 3]
```diff
@@ -73,60 +73,43 @@ def build_courses(html: str, course_names: dict) -> list[Course]:
...
-                    day=day,
-                    start_time=timings[index],
-                    location=location,
-                    duration=cell_duration
-                )
-
-        # Day ended, add the last class
-        if prev is not None:
+                # reinstantiate the prev course
+                prev = Course(code=code, name=course_names.get(code, ""), day=day,
```
Comment (harshkhandeparkar) [inline timetable/extractor.py:106]: have you tested this new code?

**Review by harshkhandeparkar — COMMENTED** [review]
just have some questions, looks good otherwise.

File: timetable/extractor.py (hunk 2 above)
Comment (yashsirvi) [reply timetable/extractor.py:89]: alright

File: timetable/extractor.py (hunk 3 above)
Comment (yashsirvi) [reply timetable/extractor.py:106]: I have tested it on my timetable, and it works. Please test it once on other timetables as well

**Review by harshkhandeparkar — APPROVED** [review]

**Pull request merged**